src/main/java/net/fabricmc/loom/processors/aw/AccessWidenerJarProcessor.java

```diff
@@ -50,98 +51,73 @@
+		Files.createDirectories(to.getParent());
+		Files.copy(from, to);
 
-				AccessWidenerRemapper remapper = new AccessWidenerRemapper(accessWidener, tinyRemapper.getRemapper(), "named");
-				accessWidener = remapper.remap();
+		ZipUtil.transformEntries(to.toFile(), getTransformers(project.getLogger(), accessWidener));
+		ZipUtil.addEntry(to.toFile(), "aw.sha256", Checksum.sha256(file));
 
-				tinyRemapper.finish();
-			} catch (IOException e) {
-				throw new RuntimeException("Failed to remap access widener", e);
-			}
-		}
+		return false;
 	}
 
 	@Override
-	public void process(File file) {
-		project.getLogger().lifecycle("Processing file: " + file.getName());
-		ZipUtil.transformEntries(file, getTransformers(accessWidener.getTargets()));
-		ZipUtil.addEntry(file, "aw.sha256", inputHash);
-	}
+	public void processRemapped(Project project, RemapJarTask task, Remapper asmRemapper, Path jar) throws IOException {
+		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
+		AccessWidener widener;
 
-	private ZipEntryTransformerEntry[] getTransformers(Set<String> classes) {
-		return classes.stream()
-				.map(string -> new ZipEntryTransformerEntry(string.replaceAll("\\.", "/") + ".class", getTransformer(string)))
-				.toArray(ZipEntryTransformerEntry[]::new);
-	}
-
-	private ZipEntryTransformer getTransformer(String className) {
-		return new ByteArrayZipEntryTransformer() {
-			@Override
-			protected byte[] transform(ZipEntry zipEntry, byte[] input) {
-				ClassReader reader = new ClassReader(input);
-				ClassWriter writer = new ClassWriter(0);
-
-				project.getLogger().lifecycle("Applying access widener to " + className);
+		if (extension.accessWidener == null) {
+			return;
+		}
 
-				reader.accept(new AccessTransformer(writer), 0);
-				return writer.toByteArray();
-			}
-		};
-	}
+		try (BufferedReader reader = new BufferedReader(new FileReader(extension.accessWidener))) {
+			widener = new AccessWidener(reader);
+		} catch (IOException e) {
+			throw new RuntimeException("Failed to read project access widener file");
+		}
 
-	//Called when remapping the mod
-	public void remapAccessWidener(Path modJarPath, Remapper asmRemapper) throws IOException {
-		AccessWidenerRemapper remapper = new AccessWidenerRemapper(accessWidener, asmRemapper, "intermediary");
-		AccessWidener remapped = remapper.remap();
+		AccessWidener remapped = AccessWidenerRemapper.remap(widener, asmRemapper, "intermediary");
 
 		StringWriter writer = new StringWriter();
 		remapped.write(writer);
 		byte[] bytes = writer.toString().getBytes();
 		writer.close();
 
-		String path = getAccessWidenerPath(modJarPath);
+		String path = getAccessWidenerPath(jar);
 
 		if (path == null) {
-			throw new RuntimeException("Failed to find accessWidener in fabric.mod.json");
+			throw new RuntimeException("Could not find an accessWidener entry in fabric.mod.json");
```

Comment:
Is there some way to say what access widener entry you were trying to find?

Reply:
What do you mean? This is on the final mapped jar and if they are using access wideners it finds the location to place the mapped access widener

Reply:
Specify in the error message what entry could not be found

Reply:
https://github.com/FabricMC/fabric-loom/commit/127e7a43304c70ebb7028134c21176fa98d872bc